src/data/polynomial/coeff.lean

```diff
@@ -61,6 +62,13 @@ lemma coeff_mul (p q : polynomial R) (n : ℕ) :
   coeff (p * q) n = ∑ x in nat.antidiagonal n, coeff p x.1 * coeff q x.2 :=
 add_monoid_algebra.mul_apply_antidiagonal p q n _ (λ x, nat.mem_antidiagonal)
 
+lemma coeff_mul' (n : ℕ) (p q : polynomial R) [comm_monoid R] :
+  coeff (p * q) n = ∑ i in range (n + 1), coeff p i * coeff q (n - i) :=
+begin
+  rw coeff_mul,
+  convert ← nat.sum_antidiagonal_eq_sum_range_succ (λ a, λ b, coeff p a * coeff q b) n,
```

Comment:
Do you even need `convert` here? Does `rw nat.sum_antidiagonal_eq_sum_range_succ` do the trick?

If so, I'd be tempted to remove this lemma entirely, and just do the two rewrites where you use it.

Reply:
Ah, `rw` does work. Since this is so short now, I agree there's no real need for this lemma. Although it might be better to keep it for visibility? Personally, I missed `sum_antidiagonal_eq_sum_range_succ`, which was the reason I wrote this in the first place.

Reply:
There's also the fact that summing over `range (n + 1)` is probably the more "normal" form, compared to `coeff_mul`.

Reply:
Right, but the antidiagonal form has the bonus of not using nat subtraction which can be awkward - plus it's more symmetric in the two indices.

What would have made that lemma easier to find for you? Is there another docstring we could have linked it from.

Reply:
Ok, I'll remove the lemma and add some doc-strings that reference `sum_antidiagonal_eq_sum_range_succ` as a way to transform the sum into its `range` form.